crates/matrix-sdk/src/event_cache/mod.rs

```diff
@@ -319,6 +319,12 @@ impl EventCache {
 
         let room_cache = self.inner.for_room(room_id).await?;
 
+        // If the linked chunked already has at least one chunk (gap or events), ignore
+        // this request, as it should happen at most once per room.
+        if room_cache.inner.state.read().await.events().chunks().next().is_some() {
```

Comment:
because the linkedchunk has, by default, one chunk, wouldn't trigger false-positive here?

Reply:
Duh, I think you're right, and this effectively disables the hacky temporary cache. And since I've removed `add_initial_events` from our testing recently, we didn't notice. Oh well, will fix today :)